### DISCUSSION TITLE
automatic APP_URL setting

### DISCUSSION THREAD

**Discussion by fariouche** [LycheeOrg/Lychee#2530] [Q&A] [answered]
Hello,
up until now, I always set APP_URL to the default value http://localhost.
I've migrated to version 5.5.1 and now APP_URL must be set to the real domain explicitly and make my picture not showing anymore if I keep the old value.

My issue here is that my server is both accessible from my LAN and the internet and the domains are not the same.
Why this check now, since the server know the domain of the request since it can display it in the diagnostic window?
What can I do?

Thanks!

**Comment by fariouche** [comment]
any idea on how i can solve this issue?

**Reply by d7415** [reply]
> any idea on how i can solve this issue?

The simplest may just be to use the public URL and add it to your hosts file if you can't use that from inside the LAN.

**Comment by fariouche** [comment]
It seems like I don't have any other choice. The public url is protected by a reverse proxy that forces mTLS, so not ideal.

Why was the behavior changed by the way? Is it for a specific security issue ?

**Reply by d7415** [reply, marked as answer]
Hosts file might get around that.

I believe it was a change because of the Livewire front end, but I'm not 100% on that. If so it *may* change back for v6 when that happens.

**Comment by fariouche** [comment]
Thanks,
Livewire never worked for me (503 errors) so I switched back :sweat_smile: 
Thanks for your time!
